src/composer/src/build.rs

```diff
@@ -284,11 +332,31 @@ fn comp_gen_make_cmd(
             (false, ifpath)
         });
 
+
     let mut optional_cmds = String::from("");
     optional_cmds.push_str(&format!("COMP_INITARGS_FILE={} ", args_file));
     if let Some(s) = tar_file {
         optional_cmds.push_str(&format!("COMP_TAR_FILE={} ", s));
     }
+
+    if !c.constants.is_empty() {
+        let mut header_content = String::from("#ifndef COMPONENT_CONSTANTS_H\n#define COMPONENT_CONSTANTS_H\n\n");
```

Comment:
The construction of the file contents should be in the following `if`, right? No reason to do all of this if a header path isn't passed in.

Reply:
make sense, will change it